data/src/main/scala/ch.epfl.scala.index.data/bintray/DownloadPoms.scala

```diff
@@ -1,90 +1,130 @@
-      broadcast.out(1).filter(hostIs("dl.bintray.com"))     .via(dlBintrayHttpFlow)      ~> merge.in(1)
-      FlowShape(broadcast.in, merge.out)
-    })
-
-  private val source = scala.io.Source.fromFile(bintrayCheckpoint.toFile)
-  private val searchesBySha1 = BintrayMeta.sortedByCreated(bintrayCheckpoint).
-    filter(s => !Files.exists(pomPath(s))). // todo make sure file content matches sha1!
-    groupBy(_.sha1).                        // remove duplicates with sha1
-    map{case (_, vs) => vs.head}
-
-  private val progress = new ProgressBar("Downloading POMs", searchesBySha1.size)
-  
-  private def pomPath(search: BintraySearch) = bintrayPomBase.resolve(s"${search.sha1}.pom")
-  
-  private val downloadPoms = Source(searchesBySha1)
-    .map(s => (download(s), s))
-    .via(splitConnectionPool)
-    .mapAsync(1){
-      case (Success(response), search) => {
-        Unmarshal(response).to[String].map(body =>
-          if(response.status == StatusCodes.OK) Right((body, search))
-          else Left((body, search))
-        )
+
+  /**
+   * verify the sha1 file hash
+   * @param path the path to the file to verify
+   * @param sha1 the sha1 hash
+   * @return
+   */
+  private def verifySHA1FileHash(path: Path, sha1: String): Boolean = {
+
+    val source = scala.io.Source.fromFile(path.toFile)
+    val content = source.mkString.split(nl).mkString("")
+    source.close()
+
+    verifyChecksum(content, sha1)
+  }
+  /**
+   * get a list of bintraySearch object where no sha1 file exists
+   */
+  private val searchesBySha1: Set[BintraySearch] = {
+
+    BintrayMeta.readQueriedPoms(bintrayCheckpoint)
+      .filter(s => !Files.exists(pomPath(s)) || !verifySHA1FileHash(pomPath(s), s.sha1))
+      .groupBy(_.sha1)                         // remove duplicates with sha1
+      .map { case (_, vs) => vs.head }
+      .toSet
+  }
+
+  /**
+   * partly url encode - replaces only spaces
+   *
+   * @param url the url to "encode"
+   * @return
+   */
+  private def escape(url: String) = url.replace(" ", "%20")
+
+  /**
+   * get the download request object
+   * downloads pom from
+   * - jcenter.bintray.com or
+   * - dl.bintray.com
+   *
+   * @param search the bintray Search
+   * @return
+   */
+  private def downloadRequest(search: BintraySearch): WSRequest = {
+
+    if (search.repo == "jcenter" && search.owner == "bintray") {
+
+      wsClient.url(escape(s"https://jcenter.bintray.com/${search.path}"))
+    } else {
+
+      wsClient.url(escape(s"https://dl.bintray.com/${search.owner}/${search.repo}/${search.path}"))
+    }
+  }
+
+  /**
+   * handle the downloaded pom and write it to file
+   * @param search the bintray search
+   * @param response the download response
+   */
+  private def processPomDownload(search: BintraySearch, response: WSResponse): Unit = {
+
+    if (200 == response.status) {
+
+      val path = pomPath(search)
+
+      if (Files.exists(path)) {
```

Comment:
We do have some equivalent contents, like two paths giving the same sha, that's allright we can keep one or the other since they are equivalent :-)